shop/src/data/state.js

```diff
@@ -352,20 +352,16 @@ const reducer = (state, action) => {
   const taxRate = parseFloat(get(newState, 'cart.taxRate', 0))
 
   const discountObj = get(newState, 'cart.discountObj', {})
-  const { minCartValue, maxDiscountValue, discountType } = get(
-    newState,
-    'cart.discountObj',
-    {}
-  )
+  const { minCartValue, maxDiscountValue, discountType } = discountObj
 
   let discount = 0
 
-  if (!minCartValue || newState.cart.subTotal > minCartValue) {
+  const totalWithShipping = get(newState, 'cart.subTotal', 0) + shipping
+  if (!minCartValue || totalWithShipping > minCartValue * 100) {
```

Comment:
Maybe we should move all the cart calculation code that is shared between backend and front-end to packages/utils/cart.js so that we don't get into a situation where the back-end cart calc does not match the ones from the front-end.

Not a must have in this PR we can add a TODO / handle in a separate PR.

Reply:
Yep, agreed